### PR TITLE
 Create DebugRequestActivityLogger and ProductionRequestActivityLogge…

### PR DESCRIPTION
**Pull request by maiklgravi**
…r for extrass data;

### CODE REVIEW

**Review by egorbwork — CHANGES_REQUESTED** [review]
Unfornutaley changes contains numerous mistakes.

File: app/Http/Controllers/FilmController.php [hunk 1]
```diff
@@ -23,11 +25,13 @@ public function index ()
         return view('film.film',['films'=>$films , 'genre'=>$genre]);
 
     }
-    public function show ($articlesId) 
+    public function show ($articlesId, Request $request, ModelLogger $logger)
```
[the remark this answers is not in the document]
Comment (maiklgravi) [reply app/Http/Controllers/FilmController.php]: I changed this.

File: app/Http/Controllers/FilmController.php [hunk 2]
```diff
@@ -23,11 +25,13 @@ public function index ()
...
-    public function show ($articlesId) 
+    public function show ($articlesId, Request $request, ModelLogger $logger) 
     {   
-        $films = Film::findOrFail($articlesId);
+        
+        $film = Film::findOrFail($articlesId);
+        $logger->logModel($request->user(),$film );
         return view('film_item' , [
-            'films' => $films , 
+            'films' => $film ,
```
[the remark this answers is not in the document]
Comment (maiklgravi) [reply app/Http/Controllers/FilmController.php]: I change this

File: app/Models/Film.php [hunk 3]
```diff
@@ -26,6 +26,20 @@ public function filmsComments(){
     public function filmsLike(){
         return $this->hasMany(FilmsLike::class);
     }
+    public function convertToLoggableString(): string{
+        return "Atriсle with id {$this->id}";
```
[the remark this answers is not in the document]
Comment (maiklgravi) [reply app/Models/Film.php]: I rename this.

File: app/Services/DebugRequestActivityLogger.php [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+use Illuminate\Http\Request;
+use Psr\Log\LoggerInterface;
+
+class DebugRequestActivityLogger extends AbstractRequestActivityLogger
+{
+    protected function collectRequestData(Request $request): array
+    {
+        return (
+            "Has oppen film with id" . $request->id
```
Comment (egorbwork) [inline app/Services/DebugRequestActivityLogger.php:12]: Should be associative array, provided string is invalid solution.

Comment (egorbwork) [reply app/Services/DebugRequestActivityLogger.php:12]: Also should not be used request input data.

Comment (egorbwork) [reply app/Services/DebugRequestActivityLogger.php:12]: Same problem in next logger. Read documentation, to see what important information you can extract from Request.

**Review by egorbwork — CHANGES_REQUESTED** [review]
Logging of model were fixed, but logger for requests are still invalid.

**Review by egorbwork — APPROVED** [review]
Can be merged.

**Pull request merged**